lib/view_model/active_record/update_data.rb

```diff
@@ -90,71 +341,86 @@ module Schemas
                                                       FunctionalUpdate::Update::NAME,
                                                       FunctionalUpdate::Remove::NAME] },
             VALUES_ATTRIBUTE => { 'type'  => 'array',
-                                  'items' => ViewModel::Schemas::VIEWMODEL_UPDATE_SCHEMA }
+                                  'items' => value_schema }
           },
           'required' => [ViewModel::TYPE_ATTRIBUTE, VALUES_ATTRIBUTE]
         }
+      end
 
-      append = base_functional_update_schema.deep_merge(
-        {
-          'description'          => 'collection append',
-          'additionalProperties' => false,
-          'properties'           => {
-            ViewModel::TYPE_ATTRIBUTE => { 'enum' => [FunctionalUpdate::Append::NAME] },
-            BEFORE_ATTRIBUTE => viewmodel_reference_only,
-            AFTER_ATTRIBUTE  => viewmodel_reference_only
-          },
-        }
-      )
-
-      APPEND_ACTION = JsonSchema.parse!(append)
-
-      update = base_functional_update_schema.deep_merge(
-        {
-          'description'          => 'collection update',
-          'additionalProperties' => false,
-          'properties'           => {
-            ViewModel::TYPE_ATTRIBUTE => { 'enum' => [FunctionalUpdate::Update::NAME] }
-          },
-        }
-      )
-
-      UPDATE_ACTION = JsonSchema.parse!(update)
-
-      remove = base_functional_update_schema.deep_merge(
-        {
-          'description'          => 'collection remove',
-          'additionalProperties' => false,
-          'properties'           => {
-            ViewModel::TYPE_ATTRIBUTE => { 'enum' => [FunctionalUpdate::Remove::NAME] },
-            # The VALUES_ATTRIBUTE should be a viewmodel_reference, but in the
-            # name of error messages, we allow more keys and check the
-            # constraint in code.
-          },
-        }
-      )
-
-      REMOVE_ACTION = JsonSchema.parse!(remove)
-
-      collection_update =
+      append_mixin = {
+        'description'          => 'collection append',
+        'additionalProperties' => false,
+        'properties'           => {
+          ViewModel::TYPE_ATTRIBUTE => { 'enum' => [FunctionalUpdate::Append::NAME] },
+          BEFORE_ATTRIBUTE          => viewmodel_reference_only,
+          AFTER_ATTRIBUTE           => viewmodel_reference_only
+        },
+      }
+
+      fupdate_owned =
+        fupdate_base.(ViewModel::Schemas::VIEWMODEL_UPDATE_SCHEMA)
+
+      fupdate_shared           =
+        fupdate_base.({ 'oneOf' => [ViewModel::Schemas::VIEWMODEL_REFERENCE_SCHEMA,
```

Comment:
I accept that, but as it is, my current thoughts reading pretty much any part of the file is "what is this type and what does it do?". The answer is often a stack of several method calls away and not at all easy to find.